### PR TITLE
Feature/recipe overview

### PR DESCRIPTION
**Pull request by CapraSnitzel**
## Description
- [x] What has been changed?
A new "Recipe Overview" screen has been added to the application. This screen displays:
  - Recipe image and title
  - Recipe rating and category
  - Preparation time, cooking time, and total time
  - Two tabs that we switch from one another:
     - One for viewing the recipe instructions
     - Another for displaying the list of ingredients per serving, with an option to add more servings (currently, the ingredients quantity does not change when more servings are added).
- [x] Why was this change made?
This feature was added to provide users with an easy-to-read overview of a specific recipe, improving the user experience by giving them all the necessary details (image, title, rating, category, and times) at a glance, along with detailed instructions and ingredients for convenience.

### Recipe Overview Screen
Screenshot 2024-10-18 at 00 48 39

### Instructions Tab
Screenshot 2024-10-18 at 00 48 48

### Ingredients Tab
Screenshot 2024-10-18 at 00 48 57

## Checklist
- [x] Tests added.
- [ ] Documentation updated.
- [x] All CI checks passed.
- [ ] Linked issue/feature (if applicable):

### CODE REVIEW

File: app/src/main/java/com/android/sample/ui/recipeOverview/RecipeOverview.kt [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+                    // Display of the list of ingredients with the ability to change the number of
+                    // servings
+                    IngredientInstructionView(ingredientsView, servingsCount, currentRecipe) {
+                        newServingsCount ->
+                      servingsCount = newServingsCount
+                    }
+                  }
+            }
+      }
+}
```
Comment (RagdollGoDown) [inline app/src/main/java/com/android/sample/ui/recipeOverview/RecipeOverview.kt:156]: We could extract some of the code in a function.

**Review by RagdollGoDown — CHANGES_REQUESTED** [review]
The PR will be a great addition to our project. I think it's practically done. It's already well separated into smaller parts. Although, would it be possible to make the RecipeOverview function smaller?

We also need to merge with main's changes.

File: app/src/main/java/com/android/sample/ui/recipeOverview/RecipeOverview.kt (hunk 1 above)
Comment (CapraSnitzel) [reply app/src/main/java/com/android/sample/ui/recipeOverview/RecipeOverview.kt:156]: Done!

**Review by RagdollGoDown — APPROVED** [review]
This all looks good. Later we need to take care of the coverage.

**Pull request merged**